web/modules/custom/collection/src/Plugin/Block/JoinCollectionBlock.php

```diff
@@ -56,12 +57,13 @@ class JoinCollectionBlock extends BlockBase implements ContainerFactoryPluginInt
    * @param \Drupal\Core\Session\AccountProxyInterface $user
    *   The current user.
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition, RouteMatchInterface $current_route_match, AccountProxyInterface $user) {
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, RouteMatchInterface $current_route_match, AccountProxyInterface $user, ContextProviderInterface $collection_context) {
     parent::__construct($configuration, $plugin_id, $plugin_definition);
     $this->currentRouteMatch = $current_route_match;
     $this->user = $user;
-    // Retrieve the collection from the route.
-    $this->collection = $this->currentRouteMatch->getParameter('rdf_entity');
+    if (!empty($collection_context->getRuntimeContexts(['collection'])['collection']->getContextValue())) {
+      $this->collection = $collection_context->getRuntimeContexts(['collection'])['collection']->getContextValue();
+    }
```

Comment:
The collection is required for this to work, it cannot be skipped. We are using the context for the block visibilty, so if this doesn't exists then we have a serious problem. If you want to do a check here, then at least throw an exception when the check fails.